File: examples/python/example_threshold.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-  

HOST = "localhost"
PORT = 4223
UID = "XYZ" # Change to your UID

from tinkerforge.ip_connection import IPConnection
from tinkerforge.bricklet_industrial_dual_analog_in import IndustrialDualAnalogIn

# Callback for voltage greater than 10V
def cb_reached(channel, voltage):
    print('Voltage (channel ' + str(channel) + ') is greater than 10V: ' + str(voltage/1000.0))

if __name__ == "__main__":
    ipcon = IPConnection() # Create IP connection
    dual_ain = IndustrialDualAnalogIn(UID, ipcon) # Create device object

    ipcon.connect(HOST, PORT) # Connect to brickd
    # Don't use device before ipcon is connected

    # Get threshold callbacks with a debounce time of 10 seconds (10000ms)
    dual_ain.set_debounce_period(10000)

    # Register threshold reached callback to function cb_reached
    dual_ain.register_callback(dual_ain.CALLBACK_VOLTAGE_REACHED, cb_reached)

    # Configure threshold (channel 0) for "greater than 10V" (unit is mV)
    dual_ain.set_voltage_callback_threshold(0, '>', 10*1000, 0)

    raw_input('Press key to exit\n') # Use input() in Python 3
    ipcon.disconnect()

File: software/examples/vbnet/ExampleThreshold.vb
Imports System
Imports Tinkerforge

Module ExampleThreshold
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change XYZ to the UID of your Industrial Dual Analog In Bricklet

    ' Callback subroutine for voltage reached callback
    Sub VoltageReachedCB(ByVal sender As BrickletIndustrialDualAnalogIn, _
                         ByVal channel As Byte, ByVal voltage As Integer)
        Console.WriteLine("Channel: " + channel.ToString())
        Console.WriteLine("Voltage: " + (voltage/1000.0).ToString() + " V")
        Console.WriteLine("")
    End Sub

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim idai As New BrickletIndustrialDualAnalogIn(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Get threshold callbacks with a debounce time of 10 seconds (10000ms)
        idai.SetDebouncePeriod(10000)

        ' Register voltage reached callback to subroutine VoltageReachedCB
        AddHandler idai.VoltageReachedCallback, AddressOf VoltageReachedCB

        ' Configure threshold for voltage "greater than 10 V"
        idai.SetVoltageCallbackThreshold(1, ">"C, 10*1000, 0)

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module
